esmf_regrid/tests/unit/schemes/test_ESMFNearestRegridder.py

```diff
@@ -0,0 +1,318 @@
+
+    Checks that the `use_src_mask` and `use_tgt_mask` keywords work properly.
+    """
+    src = _curvilinear_cube(7, 6, [-180, 180], [-90, 90])
+    tgt = _curvilinear_cube(6, 7, [-180, 180], [-90, 90])
+
+    # Make src and tgt discontiguous at (0, 0)
+    src_mask = np.zeros([6, 7], dtype=bool)
+    src_mask[0, 0] = True
+    src.data = np.ma.array(src.data, mask=src_mask)
+    src_discontiguous = src.copy()
+    src_discontiguous.coord("latitude").bounds[0, 0] = 0
+    src_discontiguous.coord("longitude").bounds[0, 0] = 0
+
+    tgt_mask = np.zeros([7, 6], dtype=bool)
+    tgt_mask[0, 0] = True
+    tgt.data = np.ma.array(tgt.data, mask=tgt_mask)
+    tgt_discontiguous = tgt.copy()
+    tgt_discontiguous.coord("latitude").bounds[0, 0] = 0
+    tgt_discontiguous.coord("longitude").bounds[0, 0] = 0
+
+    rg_src_masked = ESMFNearestRegridder(src_discontiguous, tgt, use_src_mask=True)
+    rg_tgt_masked = ESMFNearestRegridder(src, tgt_discontiguous, use_tgt_mask=True)
+    rg_unmasked = ESMFNearestRegridder(src, tgt)
+
+    weights_src_masked = rg_src_masked.regridder.weight_matrix
+    weights_tgt_masked = rg_tgt_masked.regridder.weight_matrix
+    weights_unmasked = rg_unmasked.regridder.weight_matrix
+
+    # Check there are no weights associated with the masked point.
+    assert weights_src_masked[:, 0].nnz == 0
+    assert weights_tgt_masked[0].nnz == 0
+
+    # Check other weights are correct. Note that unique to NEAREST_DTOS, masking a source
+    # point causes the next nearest source point to gain weights in the weight matrix.
+    # because of this, we ignore the row associated with that target point and check
+    # the rest of the weights matrix.
+    assert np.allclose(weights_src_masked[1:].todense(), weights_unmasked[1:].todense())
+    assert np.allclose(weights_tgt_masked[1:].todense(), weights_unmasked[1:].todense())
+
+
+def test_regrid_data():
+    """
+    Test initialisation of :func:`esmf_regrid.schemes.ESMFNearestRegridder`.
+
+    Checks that regridding mathematics behaves in an expected way.
+    """
+    # Create two similar grids so that source data and expected result
+    # data ought to look similar by visual inspection.
+    src = _grid_cube(5, 4, [-180, 180], [-90, 90], circular=True)
+    tgt = _grid_cube(4, 5, [-180, 180], [-90, 90], circular=True)
+
+    src_curv = _curvilinear_cube(5, 4, [-180, 180], [-90, 90])
+    tgt_curv = _curvilinear_cube(4, 5, [-180, 180], [-90, 90])
+
+    src_mesh_cube = _gridlike_mesh_cube(5, 4)
+    tgt_mesh_cube = _gridlike_mesh_cube(4, 5)
+
+    src_data = np.arange(20).reshape([4, 5])
+    src.data = src_data
+    src_curv.data = src_data
+    src_mesh_cube.data = np.arange(20)
+
+    rg = ESMFNearestRegridder(src, tgt)
+    rg_g2m = ESMFNearestRegridder(src, tgt_mesh_cube)
+    rg_m2g = ESMFNearestRegridder(src_mesh_cube, tgt)
+    rg_g2c = ESMFNearestRegridder(src, tgt_curv)
+    rg_c2g = ESMFNearestRegridder(src_curv, tgt)
```

Comment:
Parameterising this isn't just about readability, but is important for showing which, if any, case is failing. I'm not especially concerned if you wanted to leave the setup steps the same, or conditional to which test is running.